### PR TITLE
[tests] Fix bug in weights tensor layout in solver test

### PR DESCRIPTION
**Pull request by xinlipn**
Weight tensor should be initilized as k, C, y, x instead of k, C, x, y

### CODE REVIEW

**Comment by junliume** [comment]
@xinlipn please add an issue where this problem is found, and add a test :)

**Comment by xinlipn** [comment]
> @xinlipn please add an issue where this problem is found, and add a test :)
Hi @junliume , this was discovered while @JehandadKhan was reviewing another PR#22 on internal branch. This itself is a test :)

**Comment by xinlipn** [comment]
Local test seems to be ok. Will try to reproduce on different nodes.

130/136 Test #130: LOG_TEST.AssertLogCmdOutput_Neg ...........................................................................................................   Passed    0.08 sec
        Start 131: LOG_TEST.AssertLogFindCmdOutput_Neg
131/136 Test #131: LOG_TEST.AssertLogFindCmdOutput_Neg .......................................................................................................   Passed    0.08 sec
        Start 132: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:16C:128H:16W:16k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
132/136 Test #132: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:16C:128H:16W:16k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed    0.25 sec
        Start 133: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:128H:28W:28k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
133/136 Test #133: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:128H:28W:28k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed    0.49 sec
        Start 134: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:256H:14W:14k:256y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
134/136 Test #134: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:256H:14W:14k:256y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed    0.38 sec
        Start 135: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:512H:7W:7k:512y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
135/136 Test #135: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:512H:7W:7k:512y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) ...........   Passed    0.41 sec
        Start 136: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:1024H:14W:14k:1024y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
136/136 Test #136: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:1024H:14W:14k:1024y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .......   Passed    1.15 sec

100% tests passed, 0 tests failed out of 85
Total Test time (real) = 127.38 sec

**Comment by xinlipn** [comment]
Tested on gfx908 node and also passed. 

        Start 133: LOG_TEST.AssertLogFindCmdOutput_Neg
133/138 Test #133: LOG_TEST.AssertLogFindCmdOutput_Neg .......................................................................................................   Passed    0.10 sec
        Start 134: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:16C:128H:16W:16k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
134/138 Test #134: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:16C:128H:16W:16k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed    1.88 sec
        Start 135: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:128H:28W:28k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
135/138 Test #135: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:128H:28W:28k:128y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed   13.49 sec
        Start 136: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:256H:14W:14k:256y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
136/138 Test #136: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:256H:14W:14k:256y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .........   Passed   12.31 sec
        Start 137: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:512H:7W:7k:512y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)
137/138 Test #137: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:512H:7W:7k:512y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) ...........   Passed   11.86 sec
        Start 138: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:1024H:14W:14k:1024y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0)

138/138 Test #138: ConvFwdTest/ConvFwdSolverTestFloat.ConvASM3x3UFwd/(1,N:64C:1024H:14W:14k:1024y:3x:3pad_y:1pad_x:1stride_y:1dilation_y:1conv_mode:0) .......   Passed  178.18 sec

100% tests passed, 0 tests failed out of 87

Total Test time (real) = 393.76 sec

**Comment by atamazov** [comment]
@xinlipn Please format logs like this:
````
```
log text (will be rendered in monospaced font, no word wrapping)
```
````
When logs are very long, either attach as text files or consult https://github.com/ROCmSoftwarePlatform/MIOpen/wiki/How-to-insert-console-logs-into-github-pages

**Comment by atamazov** [comment]
@xinlipn Please add `[tests]` to the title of this PR, thanks.

File: test/gtest/solver.hpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ struct ConvFwdSolverTest
         test_skipped                = false;
         std::tie(algo, conv_config) = GetParam();
         input   = tensor<T>{conv_config.N, conv_config.C, conv_config.H, conv_config.W};
-        weights = tensor<T>{1, conv_config.k, conv_config.x, conv_config.y};
+        weights = tensor<T>{conv_config.k, conv_config.C, conv_config.x, conv_config.y};
```
Comment (JehandadKhan) [inline test/gtest/solver.hpp]:
```suggestion
        weights = tensor<T>{conv_config.k, conv_config.C, conv_config.y, conv_config.x};
```

**Review by JehandadKhan — CHANGES_REQUESTED** [review]

File: test/gtest/solver.hpp (hunk 1 above)
Comment (xinlipn) [reply test/gtest/solver.hpp]: @JehandadKhan  done

**Comment by xinlipn** [comment]
The original file was renamed. A new PR had been created, reviewed and merged. Thus close this PR

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2436

**Pull request closed without merging**